### PR TITLE
DOC: fix PR02 errors in docstring for pandas.tseries.offsets.DateOffset

### PR DESCRIPTION
**Pull request by jordan-d-murphy**
All PR02 errors resolved for the following cases: 

1. scripts/validate_docstrings.py --format=actions --errors=PR02 pandas.tseries.offsets.DateOffset

- [x] xref https://github.com/pandas-dev/pandas/issues/57111
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jordan-d-murphy** [comment]
pre-commit.ci autofix

**Comment by jordan-d-murphy** [comment]
pre-commit.ci autofix

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -1586,8 +1586,8 @@ class DateOffset(RelativeDeltaOffset, metaclass=OffsetMeta):
     Besides, adding a DateOffsets specified by the singular form of the date
     component can be used to replace certain component of the timestamp.
 
-    Parameters
-    ----------
+    Parameters ''' numpydoc ignore=PR02 '''
+    ---------- ''' numpydoc ignore=PR02 '''
```
Comment (rhshadrach) [inline pandas/_libs/tslibs/offsets.pyx:1590]: I think I'd prefer to have the ignore in the code_checks rather than the docstring, but no strong opinion.

Comment (jordan-d-murphy) [reply pandas/_libs/tslibs/offsets.pyx:1590]: yeah I'm in agreement, these changes turned out to fail some of the CI checks, and I've been unable to get the inline comments to pass all the checks. Until we can come up with a solution to resolve the PR02s I am in agreement we should leave all the "ignore" related code in code_checks.sh

**Pull request closed without merging**